botocore/parsers.py

```diff
@@ -694,10 +695,24 @@ def _do_error_parse(self, response, shape):
         error['Error']['Message'] = body.get(
             'message', body.get('Message', '')
         )
+        # Error response may contain an x-amzn-query-error header for json
+        # we need to fetch the error code from this header in that case
+        query_error = header.get('x-amzn-query-error') if header else None
```

Comment:
Are we trying to control for a case where header isn't a dict? Is that possible here? I'd expect something more idiomatic like this:
```suggestion
        query_error = headers.get('x-amzn-query-error', '')
```

Reply:
Thanks. Changes made accordingly.